packages/toc/src/utils/headings.ts

```diff
@@ -94,59 +93,10 @@ const isNotebookHeading = (heading: any): heading is INotebookHeading => {
   return heading.type !== undefined && heading.cellRef !== undefined;
 };
 
-/**
- * Runs runnable code cells.
- *
- * @private
- * @param headings - list of headings
- * @param heading - heading
- */
-const runNestedCodeCells = (headings: IHeading[], heading: IHeading) => {
-  let h: INotebookHeading;
-  let i: number;
-
-  if (!isNotebookHeading(heading)) {
-    return;
-  }
-
-  let runCode: INotebookHeading[] = [];
-  // Find the heading in the list of headings...
-  i = headings.indexOf(heading);
-
-  // Check if the current heading is a "code" heading...
-  h = heading as INotebookHeading;
-  if (h.type === 'code') {
-    runCode.push(h);
-  } else {
-    // Check for nested code headings...
-    const level = heading.level;
-    for (i = i + 1; i < headings.length; i++) {
-      h = headings[i] as INotebookHeading;
-      if (h.level <= level) {
-        break;
-      }
-      if (h.type === 'code') {
-        runCode.push(h);
-      }
-    }
-  }
-
-  // Run each of the associated code cells...
-  for (i = 0; i < runCode.length; i++) {
-    if (runCode[i].cellRef) {
-      const cell = runCode[i].cellRef as CodeCell;
-      const panel = cell.parent?.parent as NotebookPanel;
-      if (panel) {
-        void CodeCell.execute(cell, panel.sessionContext);
-      }
-    }
-  }
-};
-
 /**
  * Exports.
  */
 export { IHeading };
 export { INumberedHeading };
 export { INotebookHeading };
-export { runNestedCodeCells };
```

Comment:
Should we keep this function (and its export) in this backport for backward compatibility?

Reply:
Good point!! let me see how to add those back

Reply:
Added them back manually :)

Reply:
Thanks!